Reorganize the output of "kubectl get -o json"

CaoShuFeng opened the pull request:
"resourceVersion" and "selfLink" are members of "metadata" rather than
the list. This change reorganize the struct to provide consistency with
other places.

Before this change:
```json
    "kind": "List",
    "metadata": {},
    "resourceVersion": "",
    "selfLink": ""
```
After this change:
```json
    "kind": "List",
    "metadata": {
        "resourceVersion": "",
        "selfLink": ""
    }
```

Reference:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L57

<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#pull-request-process and developer guide https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://github.com/kubernetes/community/blob/master/contributors/devel/faster_reviews.md
3. Follow the instructions for writing a release note: https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#release-notes
-->

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access) 
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`. 
-->
```release-note
NONE
```

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/43896)
<!-- Reviewable:end -->

CaoShuFeng committed:
Reorganize the output of "kubectl get -o json"

"resourceVersion" and "selfLink" are members of "metadata" rather than
the list. This change reorganize the struct to provide consistency with
other places.

Before this change:
    "kind": "List",
    "metadata": {},
    "resourceVersion": "",
    "selfLink": ""
After this change:
    "kind": "List",
    "metadata": {
        "resourceVersion": "",
        "selfLink": ""
    }

Reference:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L57

kargakis commented:
/lgtm

CaoShuFeng commented:
@k8s-bot non-cri e2e test this

kargakis commented:
@k8s-bot non-cri e2e test this https://github.com/kubernetes/kubernetes/issues/37435

CaoShuFeng commented:
rebased

kargakis commented:
@kubernetes/sig-cli-misc any backwards compatibility concerns?

deads2k commented:
/lgtm

deads2k commented:
> @kubernetes/sig-cli-misc any backwards compatibility concerns?

It was just bugged before.

CaoShuFeng commented:
@k8s-bot node e2e test this

kargakis commented:
@k8s-bot node e2e test this https://github.com/kubernetes/kubernetes/issues/43794

kargakis commented:
@k8s-bot gce etcd3 e2e test this https://github.com/kubernetes/kubernetes/issues/44541
@k8s-bot non-cri node e2e test this
